watertap/property_models/seawater_prop_pack.py

```diff
@@ -1447,6 +1449,26 @@ def rule_dh_vap_mass(
 
         self.eq_dh_vap_mass = Constraint(rule=rule_dh_vap_mass)
 
+    def _bpe_phase(self):
+        self.bpe_phase = Var(
+            self.params.phase_list,
+            initialize=5e-1,
+            bounds=(0, 1e9),
+            units=pyunits.K,
+            doc="Boiling point elevation",
+        )
+
+        def rule_bpe_phase(
+            b,
+        ):  # boiling point elevation of seawater from eq. 36 in Sharqawy et al. (2010)
+            t = b.temperature - 273.15 * pyunits.K
+            s = b.mass_frac_phase_comp["Liq", "TDS"]
+            A = b.params.bpe_A0 + b.params.bpe_A1 * t + b.params.bpe_A2 * t**2
+            B = b.params.bpe_B0 + b.params.bpe_B1 * t + b.params.bpe_B2 * t**2
+            return b.bpe_phase["Liq"] == A * s**2 + B * s
+
+        self.eq_bpe_phase = Constraint(rule=rule_bpe_phase)
```

Comment:
Due to some underlying issues in Pyomo, there can be cases where construction of properties on demand fails on the first attempt (due to order of operations). These instances are rare, but to protect against this you need to wrap all of this in a `try/except` block:

```
def _bpe_phase(self):
    try:
        # Build the variable and constraints
    except AttributeError:
        self.del_component(self.bpe_phase)
        self.del_component(self.eq_bpe_phase)
```

Reply:
We would need to apply this block to all other on-demand variables/constraints for consistency then. I think that is beyond the scope of this PR and can be reserved for a subsequent PR.

Reply:
The variable name has been updated to `boiling_point_elevation_phase`
Also, I noticed that for most of the variables with a phase index, the scaled constraints are not indexed (except for osmotic pressure: `pressure_osm_phase`, which has a unique way of transforming constraints). We might want to unify the transforming approach for the same type of variables as well in a subsequent PR.

@andrewlee94 For this PR I think it makes sense to just implement the BPE equation, and in a different PR we can update the approaches you mentioned. Thanks again for those insights.
@adam-a-a Thanks for helping resolve the accident merge issue.